contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -18,14 +19,33 @@
 from pex.interpreter import PythonInterpreter
 from pex.pex import PEX
 from pex.pex_info import PexInfo
+from pants.task.lint_task_mixin import LintTaskMixin
+from typing import List
 
 
-class MypyTask(ResolveRequirementsTaskBase):
-  """Invoke the mypy static type analyzer for Python."""
+class MypyTaskError(TaskError):
+  """Indicates a TaskError from a failing MyPy run."""
+
+class MypyTask(LintTaskMixin, ResolveRequirementsTaskBase):
+  """Invoke the mypy static type analyzer for Python.
+
+  Mypy lint task filters out target_roots that are not properly tagged according to
+  --whitelisted-tag-name (defaults to None, and no filtering occurs if this option is 'None'),
+  and executes MyPy on targets in context from whitelisted target roots.
+  Next, if any transitive targets from the filtered roots are not whitelisted, a Warning.
```

Comment:
nit `Warning.` -> `warning`

Reply:
And warnings aren't "thrown": ... "rendered"/"printed" maybe?